client/src/main/java/tds/assessment/Segment.java

```diff
@@ -138,70 +157,112 @@ public Instant getFieldTestStartDate() {
         return fieldTestStartDate;
     }
 
+    public void setFieldTestStartDate(Instant fieldTestStartDate) {
+        this.fieldTestStartDate = fieldTestStartDate;
+    }
+
     /**
      * @return the field test end date
      */
     public Instant getFieldTestEndDate() {
         return fieldTestEndDate;
     }
 
-    public void setSegmentId(String segmentId) {
-        this.segmentId = segmentId;
+    public void setFieldTestEndDate(Instant fieldTestEndDate) {
+        this.fieldTestEndDate = fieldTestEndDate;
     }
 
-    public void setSelectionAlgorithm(Algorithm selectionAlgorithm) {
-        this.selectionAlgorithm = selectionAlgorithm;
-    }
+    /**
+     * @return All the {@link tds.assessment.Form}s associated with this segment.
+     * <p>
+     *     This getter/setter is only used by {@link tds.assessment.Segment}s that are configured for the "fixedform"
+     *     algorithm.  Segments configured for the "adaptive2" algorithm do not have forms, meaning this can return
+     *     null.
+     * </p>
+     */
+    public List<Form> getForms() {
 
-    public void setStartAbility(float startAbility) {
-        this.startAbility = startAbility;
-    }
+        if (forms == null) {
+            forms = new ArrayList<>();
+        }
 
-    public void setSubject(String subject) {
-        this.subject = subject;
-    }
+        return forms;
 
-    public void setAssessmentKey(String assessmentKey) {
-        this.assessmentKey = assessmentKey;
-    }
-
-    public void setPosition(int position) {
-        this.position = position;
     }
 
-    public void setMinItems(int minItems) {
-        this.minItems = minItems;
+    public void setForms(List<Form> forms) {
+        this.forms = forms;
     }
 
-    public void setMaxItems(int maxItems) {
-        this.maxItems = maxItems;
-    }
+    /**
+     * Choose the {@link tds.assessment.Form} for the specified language.
+     * <p>
+     *     This method is only used by {@link tds.assessment.Segment}s that are configured for the "fixedform"
+     *     algorithm.  Segments configured for the "adaptive2" algorithm do not have forms, meaning this can return
+     *     null.
+     * </p>
+     *
+     * @param languageCode The student's specified language
+     * @return The {@link tds.assessment.Form} for the specified language
+     * @throws java.lang.IllegalArgumentException if the languageCode does not match any of the forms contained within
+     * this Segment
+     */
+    public Form getForm(String languageCode) {
```

Comment:
Unfortunately since this class is in the `client` library, returning an `Optional` will be a bit of a pain.  We could:

* Add guava (or some equivalent) library to get an `Optional` and some functional stuff (that may or may not be useful)
* Return `null`
* Return a new, empty `Form`

Also, isn't it exceptional to ask for a form with a language code the `Segment` doesn't know about?